app/browser/api/ledger.js

```diff
@@ -0,0 +1,2187 @@
+  const i = parts.path.indexOf('?')
+  if (i !== -1) {
+    parts.pathname = parts.path.substring(0, i)
+    parts.search = parts.path.substring(i)
+  } else {
+    parts.pathname = parts.path
+  }
+
+  options = {
+    url: urlFormat(parts),
+    method: params.method,
+    payload: params.payload,
+    responseType: 'text',
+    headers: underscore.defaults(params.headers || {}, {'content-type': 'application/json; charset=utf-8'}),
+    verboseP: options.verboseP
+  }
+  request.request(options, (err, response, body) => {
+    let payload
+
+    if (response && options.verboseP) {
+      console.log('[ response for ' + params.method + ' ' + parts.protocol + '//' + parts.hostname + params.path + ' ]')
+      console.log('>>> HTTP/' + response.httpVersionMajor + '.' + response.httpVersionMinor + ' ' + response.statusCode +
+        ' ' + (response.statusMessage || ''))
+      underscore.keys(response.headers).forEach((header) => {
+        console.log('>>> ' + header + ': ' + response.headers[header])
+      })
+      console.log('>>>')
+      console.log('>>> ' + (body || '').split('\n').join('\n>>> '))
+    }
+
+    if (err) return callback(err)
+
+    if (Math.floor(response.statusCode / 100) !== 2) {
+      return callback(new Error('HTTP response ' + response.statusCode) + ' for ' + params.method + ' ' + params.path)
+    }
+
+    try {
+      payload = rawP ? body : (response.statusCode !== 204) ? JSON.parse(body) : null
+    } catch (err) {
+      return callback(err)
+    }
+
+    try {
+      callback(null, response, payload)
+    } catch (err0) {
+      if (options.verboseP) console.log('\ncallback: ' + err0.toString() + '\n' + err0.stack)
+    }
+  })
+
+  if (!options.verboseP) return
+
+  console.log('<<< ' + params.method + ' ' + parts.protocol + '//' + parts.hostname + params.path)
+  underscore.keys(options.headers).forEach((header) => {
+    console.log('<<< ' + header + ': ' + options.headers[header])
+  })
+  console.log('<<<')
+  if (options.payload) console.log('<<< ' + JSON.stringify(params.payload, null, 2).split('\n').join('\n<<< '))
+}
+
+const updateLedgerInfo = (state) => {
+  const ledgerInfo = ledgerState.getInfoProps(state)
+  const now = new Date().getTime()
+
+  if (ledgerInfo.get('buyURLExpires') > now) {
+    state = ledgerState.setInfoProp(state, 'buyMaximumUSD', 6)
+  }
+  if (typeof process.env.ADDFUNDS_URL !== 'undefined') {
+    state = ledgerState.setInfoProp(state, 'buyURLFrame', true)
+    const buyURL = process.env.ADDFUNDS_URL + '?' +
+      queryString.stringify({
+        currency: ledgerInfo.get('currency'),
+        amount: getSetting(settings.PAYMENTS_CONTRIBUTION_AMOUNT),
+        address: ledgerInfo.get('address')
+      })
+    state = ledgerState.setInfoProp(state, 'buyURL', buyURL)
+    state = ledgerState.setInfoProp(state, 'buyMaximumUSD', false)
+  }
+
+  // TODO remove when BAT is implemented, we don't need this for BAT
+  /*
+  if ((client) && (now > ledgerInfo._internal.geoipExpiry)) {
```

Comment:
this should be safe to remove now since it's always in source control (if we needed it)

Reply:
done